libraries/primitive_consul_lock.rb

```diff
@@ -93,22 +95,32 @@ class Semaphore
     def self.get_or_create(path, concurrency)
       require 'diplomat'
       retry_left = 5
-      value = begin
-                Chef::Log.info "Fetch lock state for #{path}"
-                Diplomat::Kv.get(path, decode_values: true)
-              rescue Diplomat::KeyNotFound
-                initial =  { version: 1, concurrency: concurrency, holders: {} }.to_json
-                Chef::Log.info "Lock for #{path} did not exist, creating with value #{initial}"
-                Diplomat::Kv.put(path, initial, cas: 0) # we ignore success/failure of CaS
-                (retry_left -= 1) > 0 ? retry : raise
-              end.first
-              Semaphore.new(path, value)
+      value =  Mash.new({ version: 1, concurrency: concurrency, holders: {} })
+      current_lock = begin
+                       Chef::Log.info "Fetch lock state for #{path}"
+                       Diplomat::Kv.get(path, decode_values: true)
+                     rescue Diplomat::KeyNotFound
+                       Chef::Log.info "Lock for #{path} did not exist, creating with value #{value}"
+                       Diplomat::Kv.put(path, value.to_json, cas: 0) # we ignore success/failure of CaS
+                       (retry_left -= 1) > 0 ? retry : raise
+                     end.first
+      desired_lock = bootstrap_lock(value, current_lock)
+      Semaphore.new(path, desired_lock)
     end
 
-    def initialize(path, value)
+    def self.bootstrap_lock(desired_value, current_lock)
```

Comment:
FAR: IMHO this is a bit complicated to just "keep" the desired concurrency (i.e. the parameters)
The `concurrency` is computed client-side, so this is not a setting of the global lock, but the "holder" itself.